### PR TITLE
JDK15 java.lang.invoke.MemberName is reachable.

### PR DESCRIPTION
**Pull request by bobvandette**
Two JDK15 fixes needed to be able to build libjvmcicompiler.so.

1. Fix ints to long in StringConcatHelper
2. Eliminate use of MemberName in MethodHandles.ensureInitialized

### CODE REVIEW

File: substratevm/src/com.oracle.svm.core.jdk15/src/com/oracle/svm/core/jdk15/Target_java_lang_invoke_MethodHandles_JDK15OrLater.java [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+// Checkstyle: resume
+import jdk.internal.misc.Unsafe;
+
+import com.oracle.svm.core.jdk.JDK15OrLater;
+import com.oracle.svm.core.SubstrateUtil;
+import com.oracle.svm.core.annotate.Alias;
+import com.oracle.svm.core.annotate.Substitute;
+import com.oracle.svm.core.annotate.TargetClass;
+
+@TargetClass(className = "java.lang.invoke.MethodHandles", innerClass = "Lookup", onlyWith = JDK15OrLater.class)
```
Comment (christianwimmer) [inline substratevm/src/com.oracle.svm.core.jdk15/src/com/oracle/svm/core/jdk15/Target_java_lang_invoke_MethodHandles_JDK15OrLater.java:39]: I would like to have this substitution only for exactly JDK 15, and a proper fix merged into JDK 16 at the same time we merge this PR. It is not sustainable to have substitutions like this, so we need to adopt a "fix properly in master first" approach.

File: substratevm/src/com.oracle.svm.core/src/com/oracle/svm/core/jdk/Target_java_lang_StringConcatHelper.java [hunk 2]
```diff
@@ -37,6 +38,16 @@
      * replaces the allocation with a deopt.
      */
     @Substitute
+    @TargetElement(onlyWith = JDK14OrLater.class)
+    private static long checkOverflow(long len) {
+        if (len < 0) {
```
Comment (christianwimmer) [inline substratevm/src/com.oracle.svm.core/src/com/oracle/svm/core/jdk/Target_java_lang_StringConcatHelper.java:43]: You missed a subtle part of http://hg.openjdk.java.net/jdk/jdk/file/6bc1fbc6e006/src/java.base/share/classes/java/lang/StringConcatHelper.java#l54 : the conversion to `(int)` before the comparison.

This shows that my substitution-based approach for performance optimization was misguided. Please remove this whole file, we need to find a more maintainable way of doing the optimization.

File: substratevm/src/com.oracle.svm.core.jdk15/src/com/oracle/svm/core/jdk15/Target_java_lang_invoke_MethodHandles_JDK15OrLater.java (hunk 1 above)
Comment (bobvandette) [reply substratevm/src/com.oracle.svm.core.jdk15/src/com/oracle/svm/core/jdk15/Target_java_lang_invoke_MethodHandles_JDK15OrLater.java]: I've restricted the change to JDK15 and initiated a dialog with the JDK core libraries team.

File: substratevm/src/com.oracle.svm.core/src/com/oracle/svm/core/jdk/Target_java_lang_StringConcatHelper.java (hunk 2 above)
Comment (bobvandette) [reply substratevm/src/com.oracle.svm.core/src/com/oracle/svm/core/jdk/Target_java_lang_StringConcatHelper.java]: I've removed the file.  Not sure why this shows up as a rename.  I manually added JDK15OrEarlier.java??

File: substratevm/src/com.oracle.svm.core.jdk15/src/com/oracle/svm/core/jdk15/Target_java_lang_invoke_MethodHandles_JDK15OrLater.java (hunk 1 above)
Comment (bobvandette) [reply substratevm/src/com.oracle.svm.core.jdk15/src/com/oracle/svm/core/jdk15/Target_java_lang_invoke_MethodHandles_JDK15OrLater.java]: Pushed a change to JDK16 and restricted this substitution to JDK15.

File: substratevm/src/com.oracle.svm.core/src/com/oracle/svm/core/jdk/Target_java_lang_StringConcatHelper.java (hunk 2 above)
Comment (bobvandette) [reply substratevm/src/com.oracle.svm.core/src/com/oracle/svm/core/jdk/Target_java_lang_StringConcatHelper.java]: removed StringConcatHelper.

**Comment by fniephaus** [comment]
I'm going to close this PR because it targets JDK 15. The current master targets JDK 21 and later.

**Pull request closed without merging**